ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/PoWSolver.java

```diff
@@ -26,10 +26,14 @@
 
 import com.google.common.base.Stopwatch;
 import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.concurrent.ExpiringMap;
 import org.apache.tuweni.units.bigints.UInt256;
 
 public class PoWSolver {
 
+  private static final long POW_JOB_TTL = 1000 * 60 * 5; // 5 minutes
```

Comment:
is this value related to something ?

Reply:
Not really. 13s per block on average, so 13 times 8 is 108s, times 2 makes 3 minutes and a half, and buffered to 5 minutes to ensure we catch stragglers. We can use a different approach. I have another ticket to add meters to count accepted solutions vs rejected solutions.

Reply:
ok it was to make sure I didn't miss anything. did you had the opportunity to test it ? I can test it on ropsten if you want

Reply:
I did not. I kinda wanted folks to review as well. I added some unit tests to make sure it made sense.

Reply:
My instinct says this should be configurable.  Probably via an `--X` flag.

Reply:
Is 5 minutes an ok default?

Reply:
maybe 2x high?  But since we are just keeping block headers (right?) from 5 min ago unless things get _very_ hairy I don't see it causing resource problems.  If we were keeping TXes it could be problematic.

Reply:
Yep, only block headers. It should be fine.